x-pack/plugin/sql/qa/src/main/java/org/elasticsearch/xpack/sql/qa/jdbc/ResultSetTestCase.java

```diff
@@ -885,10 +894,16 @@ public void testGettingDateWithoutCalendar() throws Exception {
         Long randomLongDate = randomNonNegativeLong();
         indexSimpleDocumentWithTrueValues(randomLongDate);
         
-        doWithQuery(SELECT_ALL_FIELDS, (results) -> {
+        // Create simple timezones (in the form of Z/GMT+/-x so that timezones offsets differences between
+        // Java versions (the one server runs on and the one the test runs on) will not skew the tests' results
+        int hoursOffset = randomIntBetween(0, 18);
+        ZoneOffset randomFixedZone = ZoneOffset.ofTotalSeconds(hoursOffset * (randomBoolean() ? 1 : -1) * 3600);
```

Comment:
Don't know if you'd find relevant the timezones with sub-hour components (like Kabul +04:30, or even Australia/Eucla +08:45)

Reply:
I've added sub-hour components. Thank you for the suggestion.